controllers/datadogagent/feature/enabledefault/feature.go

```diff
@@ -0,0 +1,71 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package enabledefault
+
+import (
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+)
+
+func init() {
+	err := feature.Register(feature.DefaultIDType, buildDefaultFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func buildDefaultFeature(options *feature.Options) feature.Feature {
+	return &defaultFeature{}
+}
+
+type defaultFeature struct{}
+
+func (f *defaultFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredComponents {
+	trueValue := true
+	return feature.RequiredComponents{
+		ClusterAgent: feature.RequiredComponent{
+			IsRequired: &trueValue,
+		},
+		Agent: feature.RequiredComponent{
+			IsRequired: &trueValue,
+		},
+	}
```

Comment:
so this "feature" ensures that the agent and cluster agent are enabled if nothing else is set?

Reply:
yes,
I will add in another PR the creation of the default associate resources: serviceAccount, condigmap, RBAC...